pkg/apis/build/v1alpha1/source_types.go

```diff
@@ -184,9 +193,10 @@ func (gs *ResolvedGitSource) RevisionChanged(lastBuild *Build) bool {
 		return true
 	}
 
-	return gs.Revision != lastBuild.Spec.Source.Git.Revision
+	return gs.Commit != lastBuild.Spec.Source.Git.Revision
```

Comment:
I know why this is the way it is in this commit, but it is not pretty.

Reply:
I feel like we should still call it a revision even when it is resolved. It is not necessarily a commit?

Reply:
💯

Reply:
Are we ok with updating the json name?

Reply:
:shipit: I am

Reply:
👍 It is in the status so, it is hardly a breaking change as kpack will just regenerate it.